test/api/model/model_test.dart

```diff
@@ -628,4 +631,100 @@ void main() {
         .deepEquals([2, 3, 11]);
     });
   });
+
+  group('MessageEditState', () {
+    Map<String, dynamic> baseJson() => deepToJson(eg.streamMessage()) as Map<String, dynamic>;
+
+    test('edit history is absent', () {
+      check(Message.fromJson(baseJson()
+          ..['edit_history'] = null
+      ).messageEditState).equals(MessageEditState.none);
+
+      check(Message.fromJson(baseJson()
+          ..['edit_history'] = null
+          ..['last_edit_timestamp'] = 1678139636).messageEditState
+      ).equals(MessageEditState.edited);
+    });
+
+    test('edit history exists', () {
+      // Only channel name changed: moved
+      check(Message.fromJson(baseJson()
```

Comment:
These would make good individual test cases, i.e. individual `test` calls:
```suggestion
    group('edit history exists', () {
      test('Only channel name changed: moved', () {
        check(Message.fromJson(baseJson()
```

That finer-grained organization can become extremely helpful when making a change that breaks some of the test cases, because it makes it possible to see a full list all in one go of all the cases the change breaks, as well as the ones it doesn't.

The main cost is in vertical space occupied by the test code: it adds two lines per test case (or three lines, if they weren't already separated by blank lines). That's generally worth it when each case is already 5 lines or so or longer, as these are.